### PR TITLE
fix: crash when curl is not installed

### PR DESCRIPTION
**Pull request by bukzor**
Migrated from https://github.com/Linuxbrew/brew/issues/640

### CODE REVIEW

File: Library/Homebrew/utils/curl.rb [hunk 1]
```diff
@@ -6,8 +6,8 @@ def curl_executable
     ENV["HOMEBREW_CURL"],
     which("curl"),
     "/usr/bin/curl",
-  ].map { |c| Pathname(c) }.find(&:executable?)
-  raise "curl is not executable" unless @curl
+  ].reject(&:nil?).map { |c| Pathname(c) }.find(&:executable?)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/curl.rb]: `.reject(&:nil?)` should be `.compact`

File: Library/Homebrew/utils/curl.rb [hunk 2]
```diff
@@ -6,8 +6,8 @@ def curl_executable
...
+  raise "brew requires curl, which is not available" unless @curl
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/curl.rb]: `no executable curl was found`

File: Library/Homebrew/utils/curl.rb (hunk 1 above)
Comment (bukzor) [reply Library/Homebrew/utils/curl.rb]: My ruby-fu is quite weak. I've changed it but haven't tested.

File: Library/Homebrew/utils/curl.rb [hunk 3]
```diff
@@ -6,8 +6,8 @@ def curl_executable
     ENV["HOMEBREW_CURL"],
     which("curl"),
     "/usr/bin/curl",
-  ].map { |c| Pathname(c) }.find(&:executable?)
-  raise "curl is not executable" unless @curl
+  ].compact.map { |c| Pathname(c) }.find(&:executable?)
+  raise "brew requires curl, but no executable curl was found" unless @curl
```
[the remark this answers is not in the document]
Comment (bukzor) [reply Library/Homebrew/utils/curl.rb]: Done.

**Comment by MikeMcQuaid** [comment]
Thanks again @bukzor!

**Pull request merged**